pkg/apis/resource/types.go

```diff
@@ -0,0 +1,449 @@
+	// not yet allocated.
+	Allocation *AllocationResult
+
+	// ReservedFor indicates which entities are currently allowed to use
+	// the claim. A Pod which references a ResourceClaim which is not
+	// reserved for that Pod will not be started.
+	//
+	// The maximum size is 32.
+	ReservedFor []ResourceClaimUserReference
+
+	// DeallocationRequested gets set by the scheduler when it detects the
+	// situation where pod scheduling cannot proceed because some claim was
+	// allocated for a node that cannot provide some other required
+	// resource. It can only be set for claims that are not currently
+	// reserved.
+	//
+	// The driver then needs to deallocate this claim and reset the field
+	// together with clearing the Allocation field. The scheduler then will
+	// try again.
+	//
+	// While DeallocationRequested is set, no new users may be added to
+	// ReservedFor.
+	DeallocationRequested bool
+}
+
+// ReservedForMaxSize is the maximum number of entries in
+// claim.status.reservedFor.
+const ResourceClaimReservedForMaxSize = 32
+
+// AllocationResult contains attributed of an allocated resource.
+type AllocationResult struct {
+	// ResourceHandle contains arbitrary data returned by the driver after a
+	// successful allocation. This is opaque for
+	// Kubernetes. Driver documentation may explain to users how to
+	// interpret this data if needed.
+	//
+	// Resource drivers can use this to store some data directly or
+	// cross-reference some other place where information is stored.
+	// This data is guaranteed to be available when a Pod is about
+	// to run on a node, in contrast to the ResourceClass which
+	// may have been deleted in the meantime.
+	//
+	// The maximum size of this field is 16KiB.
+	ResourceHandle string
+
+	// This field will get set by the resource driver after it has
+	// allocated the resource driver to inform the scheduler where it can
+	// schedule Pods using the ResourceClaim.
+	//
+	// Node-local resources can use the `kubernetes.io/hostname` label
+	// to select a specific node.
+	//
+	// Setting this field is optional. If null, the resource is available
+	// everywhere.
+	AvailableOnNodes *core.NodeSelector
+
+	// ShareableResource determines whether the resource supports more
+	// than one user at a time.
+	ShareableResource bool
+}
+
+// ResourceHandleMaxSize is the maximum size of allocation.resourceHandle.
+const ResourceHandleMaxSize = 16 * 1024
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResourceClaimList is a collection of claims.
+type ResourceClaimList struct {
+	metav1.TypeMeta
+	// Standard list metadata
+	metav1.ListMeta
+
+	// Items is the list of resource claims.
+	Items []ResourceClaim
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodScheduling objects get created by a scheduler when it handles a pod which
+// uses one or more unallocated ResourceClaims with delayed allocation.
```

Comment:
Whilst that's true, the commend should explain what a PodScheduling _is_ first.

Reply:
Updated:
```
// PodScheduling objects hold information that is needed to select a node
// for a pod with ResourceClaims that use "WaitForFirstConsumer" allocation
// mode and to trigger that allocation.